core/primitives/src/sharding.rs

```diff
@@ -371,3 +375,41 @@ impl EncodedShardChunk {
         })
     }
 }
+
+/// An reed solomon instance should not consume more than 500MB of memory.
+const RS_MAX_MEMORY: u64 = 500 * 1024 * 1024;
+
+pub struct ReedSolomonWrapper {
+    rs: ReedSolomon,
+    ttl: u64,
+}
+
+impl ReedSolomonWrapper {
+    pub fn new(data_shards: usize, parity_shards: usize) -> Self {
+        ReedSolomonWrapper {
+            rs: ReedSolomon::new(data_shards, parity_shards).unwrap(),
+            ttl: RS_MAX_MEMORY / (data_shards * data_shards) as u64,
```

Comment:
Add one in case accidentally `RS_MAX_MEMORY` ends up being less than the size of a single matrix?